### PR TITLE
feat: add url-to-json

### PR DESCRIPTION
**Pull request by leozhantw**
image

### CODE REVIEW

**Comment by cloudingcity** [comment]
Please provide the actual result screenshot.

ex `https://example.com/users/1?aaa=bbb&ccc=ddd&eee=fff&aaa=ddd`

File: cmd/url.go [hunk 1]
```diff
@@ -30,3 +33,39 @@ var urlDecodeCmd = &cobra.Command{
...
+		s, err := url.Parse(args[0])
+		if err != nil {
+			fmt.Println("Invalid query")
+		}
+
+		type result struct {
+			Scheme string
+			Host   string
+			Path   string
+			Query  map[string]string
```
Comment (cloudingcity) [inline cmd/url.go:52]: I think lower case is better

File: cmd/url.go [hunk 2]
```diff
@@ -30,3 +33,39 @@ var urlDecodeCmd = &cobra.Command{
...
+			Host:   s.Host,
+			Path:   s.Path,
+		}
+
+		if s.RawQuery != "" {
+			res.Query = make(map[string]string)
+			for _, v := range strings.Split(s.RawQuery, "&") {
+				param := strings.Split(v, "=")
+				res.Query[param[0]] = param[1]
+			}
```
Comment (cloudingcity) [inline cmd/url.go:65]: You can use `Query()` method to do this

**Comment by cloudingcity** [comment]
How about doing something like this in `pkg/url` and write some unit test

```go
u := url.Parse(uri)
u.ToJSON()
```

And call this pkg in url cmd

File: cmd/url.go (hunk 2 above)
Comment (leozhantw) [reply cmd/url.go:65]: Thanks for your advice!

I have been used.

**Comment by leozhantw** [comment]
> How about doing something like this in `pkg/url` and write some unit test
> 
> ```go
> u := url.Parse(uri)
> u.ToJSON()
> ```
> 
> And call this pkg in url cmd

Ok, I have been added the pkg

**Comment by leozhantw** [comment]
> Please provide the actual result screenshot.
> 
> ex `https://example.com/users/1?aaa=bbb&ccc=ddd&eee=fff&aaa=ddd`

I have been put the result in PR description

**Comment by cloudingcity** [comment]
Thank you @leozhantw, I will release a new version for you. :smile:

**Pull request merged**